CMakeLists.txt

```diff
@@ -565,3 +575,121 @@ add_subdirectory (utils)
 include (cmake/print_include_directories.cmake)
 
 include (cmake/sanitize_target_link_libraries.cmake)
+
+# Various build speed optimizations.
+option(FASTBUILD_CONTRIB_UNITY "Use unity builds on the contrib/ folder. (No-op on CMake < 3.16.0)" ON)
+option(FASTBUILD_FUNCTIONS_UNITY "Use unity builds on the src/Functions folder. (No-op on CMake < 3.19.0)" ON)
+option(FASTBUILD_NO_PUBNAMES "Do not emit DWARF pubnames when using Clang. Slows GDB symbol loading. Reduces IO." OFF)
+option(FASTBUILD_SPLIT_DWARF "Build with split DWARF files. Makes debugging symbols non-portable. Reduces IO." OFF)
```

Comment:
Hi!
As you may know, CH has an automatic [options generator page](https://github.com/ClickHouse/ClickHouse/blob/master/docs/tools/cmake_in_clickhouse_generator.py) (builds to [this](https://clickhouse.tech/docs/en/development/cmake-in-clickhouse/)) which supports additional information that should be added as comments right before the options line, e.g.

```cmake
# Additional option data as restrictions (e.g. clang only) and other useful data.
option(OPTION "Description." OFF)
```

Some options already follow this guideline, so it would be nice if you moved the options descriptions' limitations and so into the comments.

Reply:
For example, the option FASTBUILD_CONTRIB_UNITY actually sets the unity builds for particular targets, so it's better to name them in the commens so the user would not need to look into the source code.